contracts/base/PeripheryPayments.sol

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.19;
+
+import {ERC20} from "solmate/tokens/ERC20.sol";
+import {Currency, CurrencyLibrary} from "@uniswap/v4-core/contracts/types/Currency.sol";
+import {SafeTransferLib} from "solmate/utils/SafeTransferLib.sol";
+import {IPeripheryPayments} from "../interfaces/IPeripheryPayments.sol";
+
+using CurrencyLibrary for Currency;
```

Comment:
omg i didnt know you could put this above the contract 😱 
we normally put them just inside the contract header like [this](https://github.com/Uniswap/v4-core/blob/06564d33b2fa6095830c914461ee64d34d39c305/src/PoolManager.sol#L27)

Reply:
https://docs.soliditylang.org/en/v0.8.23/contracts.html#using-for o ya i was reading this and there are some tiny differences to file level vs contract level so if we've only done contract level elsewhere it makes sense to keep that consistent xD

Reply:
wow i had no idea, ty!